packages/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,502 @@
+  name: 'KTimePicker',
+  components: {
+    KButton,
+    KPop,
+    KSegmentedControl,
+    DatePicker: () => import('v-calendar/lib/components/date-picker.umd')
+  },
+  props: {
+    defaultMessage: {
+      type: String,
+      required: false,
+      default: 'Select a time range'
+    },
+    calendarMode: {
+      type: String,
+      required: false,
+      default: '',
+      validator: (value) => {
+        return [ '', 'date', 'time', 'dateTime' ].indexOf(value) !== -1
+      }
+    },
+    defaultRelative: {
+      type: Object,
+      required: false,
+      default: () => {}
+    },
+    defaultCustom: {
+      type: Object,
+      required: false,
+      default: () => ({start: '', end: ''})
+    },
+    isRange: {
+      type: Boolean,
+      required: false,
+      default: false
+    },
+    timePeriods: {
+      type: Array,
+      required: false,
+      default: () => [],
+      validator: (valuesArray) => {
+        return valuesArray.every((item) => {
+          return Array.isArray(item.values) && item.values.every((timeframe) => {
+            // Check validity of each timeframe
+            return typeof timeframe.timeframeText === 'string' &&
+              timeframe.timeframeLength !== undefined &&
+              timeframe.start !== undefined &&
+              timeframe.end !== undefined
+          })
+        })
+      }
+    }
+  },
+
+  data () {
+    return {
+      hidePopover: false,
+      tabName: this.getDefaultTabName(),
+      abbreviatedDisplay: this.defaultMessage,
+      fullRangeDisplay: '',
+      selectedCalendarRange: this.defaultCustom,
+      selectedTimeframe: this.timePeriods[0],
+      selectedRange: { start: '', end: '' },
+      modelConfig: {
+        type: 'number'
+      },
+      calendarSelectAttributes: {
+        highlight: {
+          start: { class: 'vcal-day-start' },
+          base: { class: 'vcal-day-base' },
+          end: { class: 'vcal-day-end' },
+          day: { class: 'vcal-day-preselect' }
+        }
+      }
+    }
+  },
+
+  computed: {
+    hasCalendar () {
+      return this.calendarMode !== ''
+    },
+    hasTimePeriods () {
+      return this.timePeriods && this.timePeriods.length
+    },
+    showCalendar () {
+      return this.tabName === 'custom' || !this.hasTimePeriods
+    },
+    hasRelativeTimeframes () {
+      return this.timePeriods.length > 0
+    },
+    submitDisabled () {
+      return this.isRange || this.hasRelativeTimeframes
+        ? !this.selectedRange.start || !this.selectedRange.end
+        : !this.selectedRange.start
+    },
+    hasDefaultCustomValue () {
+      return this.defaultCustom.start !== '' && this.defaultCustom.end !== ''
+    },
+    defaultTimeframe () {
+      return (this.defaultRelative !== undefined)
+        ? this.defaultRelative
+        : null
+    }
+  },
+
+  watch: {
+    // Updates input field's "human" date whenever v-calendar value is touched
+    selectedCalendarRange (newVal) {
+      if (newVal) {
```

Comment:
question: do you also need to check here to only evaluate when `newVal !== oldVal` ?

Reply:
Yup! i just pushed that now :)